src/snowflake/cli/plugins/sql/manager.py

```diff
@@ -1,18 +1,27 @@
+from __future__ import annotations
+
 import sys
 from io import StringIO
 from pathlib import Path
-from typing import Iterable, Optional, Tuple
+from typing import Dict, Iterable, Optional, Tuple
 
-from click import UsageError
+from click import ClickException, UsageError
+from jinja2 import UndefinedError
 from snowflake.cli.api.secure_path import UNLIMITED, SecurePath
 from snowflake.cli.api.sql_execution import SqlExecutionMixin
+from snowflake.cli.api.utils.rendering import snowflake_cli_jinja_render
+from snowflake.cli.plugins.sql.snowsql_templating import transpile_snowsql_templates
 from snowflake.connector.cursor import SnowflakeCursor
 from snowflake.connector.util_text import split_statements
 
 
 class SqlManager(SqlExecutionMixin):
     def execute(
-        self, query: Optional[str], file: Optional[Path], std_in: bool
+        self,
+        query: Optional[str],
+        file: Optional[Path],
+        std_in: bool,
+        data: Dict | None = None,
```

Comment:
`Optional[Dict]`

Reply:
[PEP604](https://peps.python.org/pep-0604/) make both syntaxes compatible. We don't have a common pattern in our code base.

Reply:
`Python-Version: 3.10` What with 3.8 and 3.9?

Reply:
@sfc-gh-astus we already use it across the code base so this is nothing new. If we want to align to standard format (for example #1001 ) we should do so in other PR.